library/src/scripts/features/toaster/themePreview/ThemePreviewToastReducer.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @author Alexander Kim <alexander.k@vanillaforums.com>
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import { ILoadable, LoadStatus } from "@library/@types/api/core";
+import { produce } from "immer";
+import { reducerWithInitialState } from "typescript-fsa-reducers";
+import ThemesActions, { PreviewStatusType } from "@library/theming/ThemesActions";
+import { useSelector } from "react-redux";
+import { ICoreStoreState } from "@library/redux/reducerRegistry";
+
+export interface IPreviewToasterState {
+    applyStatus: ILoadable<{ themeID: number | string }>;
+    cancelStatus: ILoadable<{ themeID: number | string }>;
+}
+
+export interface IPreviewToasterStoreState extends ICoreStoreState {
+    themePreviewToaster: IPreviewToasterState;
+}
```

Comment:
Could we rename this file and move it up into `/theming`? I'm thinking `themePreviewReducer`

Nothing here is really specific to the toast itself.